theblues/charmstore.py

```diff
@@ -21,40 +22,39 @@
 class CharmStore(object):
     """A connection to the charmstore."""
 
-    def __init__(self, url=API_URL, macaroons=None,
-                 timeout=DEFAULT_TIMEOUT, verify=True):
+    def __init__(self, url=API_URL, timeout=DEFAULT_TIMEOUT,
+                 verify=True, client=None, cookies=None):
         """Initializer.
 
         @param url The base url to the charmstore API.  Defaults
             to `https://api.jujucharms.com/`.
-        @param macaroons The optional discharged macaroon allowing access to
-            authenticated queries against the charmstore.
         @param timeout How long to wait in seconds before timing out a request;
             a value of None means no timeout.
         @param verify Whether to verify the certificate for the charmstore API
             host.
+        @param client holds a context for making http requests with macaroons.
```

Comment:
This can be ok in Go, but in Python it does not communicate what a client should be at all, because we don't have the type information available. Maybe add one or two examples in the docstring, and please specify what types (or interfaces) are accepted here. Same for the cookies argument below.